src/lib.rs

```diff
@@ -108,38 +108,32 @@ impl FromStartEndLineCol for Diagnostic {
 // Rule trait
 // ----------
 
-/// Implemented by all rules.
-pub trait Rule {
-    fn new(settings: &Settings) -> Self
-    where
-        Self: Sized;
-}
-
 /// Implemented by rules that act directly on the file path.
-pub trait PathRule: Rule {
-    fn check(&self, path: &Path) -> Option<Diagnostic>;
+pub trait PathRule {
+    fn check(settings: &Settings, path: &Path) -> Option<Diagnostic>;
 }
 
 /// Implemented by rules that analyse lines of code directly, using regex or otherwise.
-pub trait TextRule: Rule {
-    fn check(&self, source: &SourceFile) -> Vec<Diagnostic>;
+pub trait TextRule {
+    fn check(settings: &Settings, source: &SourceFile) -> Vec<Diagnostic>;
 }
 
 /// Implemented by rules that analyse the abstract syntax tree.
-pub trait ASTRule: Rule {
-    fn check(&self, node: &Node, source: &SourceFile) -> Option<Vec<Diagnostic>>;
+pub trait ASTRule {
+    fn check(settings: &Settings, node: &Node, source: &SourceFile) -> Option<Vec<Diagnostic>>;
 
     /// Return list of tree-sitter node types on which a rule should trigger.
-    fn entrypoints(&self) -> Vec<&'static str>;
+    fn entrypoints() -> Vec<&'static str>;
 
     /// Apply a rule over some text, generating all violations raised as a result.
-    fn apply(&self, source: &SourceFile) -> anyhow::Result<Vec<Diagnostic>> {
-        let entrypoints = self.entrypoints();
+    fn apply(source: &SourceFile) -> anyhow::Result<Vec<Diagnostic>> {
+        let entrypoints = Self::entrypoints();
+        let default_settings = default_settings();
```

Comment:
I think it would be better if `apply` took settings as an argument. To avoid the need to repeatedly import `default_settings` throughout the tests, maybe we could add a convenience function of `apply_default` on `ASTRule`?

Reply:
Yes, that's a good idea. I'd like to rewrite the tests to use snapshots first as this potentially might change then anyway